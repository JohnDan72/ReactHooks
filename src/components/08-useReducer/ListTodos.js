import React from 'react';
import PropTypes from 'prop-types';
import { List } from "rsuite";
import ListTodoItem from './ListTodoItem';

const propTypes = {
    state: PropTypes.array.isRequired,
    handleTogle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
};
const defaultProps = {};

/**
 * 
 */
const ListTodos = ({ state , handleTogle , handleDelete }) => {
    return (
        <List bordered className="my-4" >
            {
                state.map(todo => (
                    <ListTodoItem   key={todo.id} todo={todo} 
                                    handleTogle={handleTogle} 
                                    handleDelete={handleDelete} />
                ))
            }

        </List>
    );
}

ListTodos.propTypes = propTypes;
ListTodos.defaultProps = defaultProps;

export default ListTodos;