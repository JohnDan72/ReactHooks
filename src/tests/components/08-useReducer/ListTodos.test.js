import { shallow } from "enzyme"
import ListTodos from "../../../components/08-useReducer/ListTodos";
import { demoTodos } from "../../fixtures/demosTodos";


describe('Pruebas ListTodos', () => {
    const [ handleTogle , handleDelete ] = [ jest.fn() , jest.fn() ];
    let wrapper = shallow(<ListTodos    state={demoTodos}
                                        handleTogle={handleTogle}
                                        handleDelete={handleDelete}/>);
    // beforeEach(() => {
    //     wrapper = shallow(<ListTodos    state={demoTodos}
    //                                     handleTogle={handleTogle}
    //                                     handleDelete={handleDelete}/>);
    // });

    test('debe renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();    
    });

    test('debe tener 3 todos list items', () => {
        const listTodos = wrapper.find('ListTodoItem');

        expect(wrapper).toMatchSnapshot();    
        expect(listTodos.length).toBe(demoTodos.length)
    });

    test('debe tener un hijo con una función llamada igual al parámetro', () => {
        const listTodos = wrapper.find('ListTodoItem');

        expect(listTodos.at(0).prop('handleDelete')).toEqual( expect.any(Function));
        expect(listTodos.at(0).prop('handleTogle')).toEqual( expect.any(Function));
    })
    
})
