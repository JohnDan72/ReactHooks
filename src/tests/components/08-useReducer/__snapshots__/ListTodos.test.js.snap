// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas ListTodos debe renderizar el componente 1`] = `
<List
  bordered={true}
  className="my-4"
>
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="2"
    todo={
      Object {
        "desc": "Aprender Angular",
        "done": false,
        "id": 2,
      }
    }
  />
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="3"
    todo={
      Object {
        "desc": "Aprender MongoDB",
        "done": false,
        "id": 3,
      }
    }
  />
</List>
`;

exports[`Pruebas ListTodos debe tener 3 todos list items 1`] = `
<List
  bordered={true}
  className="my-4"
>
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="2"
    todo={
      Object {
        "desc": "Aprender Angular",
        "done": false,
        "id": 2,
      }
    }
  />
  <ListTodoItem
    handleDelete={[MockFunction]}
    handleTogle={[MockFunction]}
    key="3"
    todo={
      Object {
        "desc": "Aprender MongoDB",
        "done": false,
        "id": 3,
      }
    }
  />
</List>
`;
